### PR TITLE
#1610 Added deny reimbursement request endpoint

### LINKED ISSUE #1610: Finance Fixes - Deny Reimbursement Request Endpoint
### Description

We want to add a denied reimbursement status type in the backend and add an endpoint that allows someone from the finance team to mark a reimbursement request as denied.

### Acceptance Criteria

- [x] A 'DENIED' option has been added to the 'Reimbursement_Status_Type' enum in the schema (you should run yarn prisma:migrate to create the migration after you've edited and saved the schema)
- [x] The shared type 'ReimbursementStatusType' has been updated to include a 'DENIED' option
- [x] Add an endpoint that adds a reimbursement_status of type denied to the specified reimbursement request
- [x] It should be on the route 'financeURL/:requestId/deny' with no payload
- [x] Only someone on the finance team should be able to deny a reimbursement request
- [x] You can't deny a 'reimbursed' request

### Proposed Solution
Look at the approveReimbursementRequest endpoint for an example

### PR DESCRIPTION
**Pull request by jessicayzhao**
## Changes

Added a denied reimbursement status type and endpoint to allow someone from the finance team to mark a reimbursement request as denied.

## Notes

Right now you can't deny a RR if its been deleted, denied, or reimbursed, not sure if should also add if approved. 

## Test Cases

- Deny Reimbursement Request fails if Submitter not on Finance Team
- Deny Reimbursement Request fails if Finance Team does not exist
- Deny Reimbursement Request fails if the Request does not exist
- Deny Reimbursement Request fails if the Request has been deleted
- Deny Reimbursement Request fails if the request has already been denied
- Deny Reimbursement Request fails if the request has already been reimbursed
- Deny Reimbursement Request success

## Checklist

It can be helpful to check the `Checks` and `Files changed` tabs.
Please review the [contributor guide](https://nerdocs.atlassian.net/wiki/spaces/NER/pages/8060929/Software+Contributor+Guide) and reach out to your Tech Lead if anything is unclear.
Please request reviewers and ping on slack only after you've gone through this whole checklist.

- [x] All commits are tagged with the ticket number
- [x] No linting errors / newline at end of file warnings
- [x] All code follows repository-configured prettier formatting
- [x] No merge conflicts
- [x] All checks passing
- [x] Screenshots of UI changes (see Screenshots section)
- [x] Remove any non-applicable sections of this template
- [x] Assign the PR to yourself
- [x] No `yarn.lock` changes (unless dependencies have changed)
- [x] Request reviewers & ping on Slack
- [x] PR is linked to the ticket (fill in the closes line below)

Closes #1610

### CODE REVIEW

File: src/backend/src/services/reimbursement-requests.services.ts [hunk 1]
```diff
@@ -700,6 +700,51 @@ export default class ReimbursementRequestService {
     return reimbursementStatusTransformer(reimbursementStatus);
   }
 
+  /**
+   * Adds a reimbursement status with type denied to the given reimbursement request
+   *
+   * @param reimbursementRequestId the id of the reimbursement request to deny
+   * @param submitter the user who is approving the reimbursement request
```
Comment (callmejim1226) [inline src/backend/src/services/reimbursement-requests.services.ts:707]: Whoops, a typo here. `approving` --> `denying`

File: src/frontend/src/utils/reimbursement-request.utils.ts [hunk 2]
```diff
@@ -59,6 +59,9 @@ export const cleanReimbursementRequestStatus = (status: ReimbursementStatusType)
     case ReimbursementStatusType.SABO_SUBMITTED: {
       return 'Submitted to Sabo';
     }
+    case ReimbursementStatusType.DENIED: {
```
Comment (callmejim1226) [inline src/frontend/src/utils/reimbursement-request.utils.ts:62]: Nice catch here!

**Review by callmejim1226 — DISMISSED** [review]
Awesome work on this new endpoint! 💯👍

**Review by RChandler234 — CHANGES_REQUESTED** [review]

**Review by RChandler234 — APPROVED** [review]
LGTM 🔥 Really good job

**Pull request merged**